src/transformers/models/clip/modeling_clip.py

```diff
@@ -571,7 +571,7 @@ class CLIPEncoder(nn.Module):
     def __init__(self, config: CLIPConfig):
         super().__init__()
         self.config = config
-        self.layers = nn.ModuleList([CLIPEncoderLayer(config) for _ in range(config.num_hidden_layers)])
+        self.layers = nn.ModuleList([thunder.jit(CLIPEncoderLayer(config)) for _ in range(config.num_hidden_layers)])
```

Comment:
This kind of update can be done in user scripts without modifying the CLIPEncoder source code. Here's an example using example code [from ModuleList docs ](https://pytorch.org/docs/stable/generated/torch.nn.ModuleList.html#torch.nn.ModuleList)
```py
from torch import nn
import torch
import thunder

class MyModule(nn.Module):
    def __init__(self):
        super().__init__()
        self.linears = nn.ModuleList([nn.Linear(10, 10) for i in range(10)])

    def forward(self, x):
        # ModuleList can act as an iterable, or be indexed using ints
        for i, l in enumerate(self.linears):
            x = self.linears[i // 2](x) + l(x)
        return x
    
model = MyModule()
model.linears = nn.ModuleList([thunder.jit(l) for l in model.linears])

x = torch.randn(10, 10)
model(x)
thunder.last_traces(model.linears[0])[-1]
```

Reply:
While true, I don't think this is feasible in the large scale, at least with current thunder. There's just too many places where a user would have to do this kind of thing instead of jitting the module as a whole. We got to modifying the transformers part of the code after multiple rounds of not taking higher-level constructs in NeMo.

Reply:
There's a system that processes Python+PyTorch code and returns plain PyTorch code for a compiler to optimize.
`@torch.compile(backend=lambda gm, _: thunder.jit(gm))` doesn't work today with a huge stack trace from Thunder's interpreter, but it was working before.